experimental/memory.go

```diff
@@ -0,0 +1,20 @@
+package experimental
+
+import (
+	"context"
+
+	"github.com/tetratelabs/wazero/internal/ctxkey"
+)
+
+type MemoryAllocator interface {
+	Make(min, cap, max uint64) []byte
+	Grow(old []byte, more uint64) []byte
+	Free([]byte)
```

Comment:
Since this is an interface, implementations can hold state, so `Grow` and `Free` could potentially _not_ receive the slice.

Also sizes are in bytes, but could just as easily be in pages. Still, page size would one more thing for implementations to mess up, so maybe this is better.